src/hwcaps.c

```diff
@@ -51,10 +52,20 @@
 // On Linux we simply use getauxval.
 #if defined(HWCAPS_REGULAR_LINUX)
 #include <dlfcn.h>
+#if defined(HAVE_SYS_AUXV_H)
 #include <sys/auxv.h>
 static unsigned long GetElfHwcapFromGetauxval(uint32_t hwcap_type) {
   return getauxval(hwcap_type);
 }
+#else // Without sys/auxv.h return 0 to fallback to GetElfHwcapFromProcSelfAuxv
+#include <elf.h>
```

Comment:
Why do you need these 3 lines?

Reply:
If you don't have sys/auxv.h then you don't have getauxval.

In this case then we should define GetElfHwcapFromGetauxval to return 0 so that GetHardwareCapabilitiesFor falls back to using GetElfHwcapFromProcSelfAuxv to find the capabilities.

elf.h is where AT_HWCAP2 will be defined if your system has it.